sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -313,11 +313,18 @@ object RapidsExecutorPlugin {
    * patch version then the actual patch version must be greater than or equal.
    * For example, version 7.1 is not satisfied by version 7.2, but version 7.1 is satisfied by
    * version 7.1.1.
+   * If the expected cudf version is a specified 'timestamp-seq' one, then it is satisfied by
+   * the SNAPSHOT version.
+   * For example, version 7.1-yyyymmdd.hhmmss-seq is satisfied by version 7.1-SNAPSHOT.
    */
   def cudfVersionSatisfied(expected: String, actual: String): Boolean = {
     val expHyphen = if (expected.indexOf('-') >= 0) expected.indexOf('-') else expected.length
     val actHyphen = if (actual.indexOf('-') >= 0) actual.indexOf('-') else actual.length
-    if (actual.substring(actHyphen) != expected.substring(expHyphen)) return false
+    if (actual.substring(actHyphen) != expected.substring(expHyphen) &&
```

Comment:
After specify a special spark-rapids-jni snapshot version jar. e.g. `-Dspark-rapids-jni.version=22.06.0-20220420.050427-10`, we would got below build info from property files

In file `rapids4spark-version-info.properties`, jni version: cudf_version=`22.06.0-20220420.050427-10`

In file `spark-rapids-jni-version-info.properties`: version=`22.06.0-SNAPSHOT`

```
root@d03af8a1b975:/spark-rapids/dist/target/tmp# cat rapids4spark-version-info.properties
version=22.06.0-SNAPSHOT
cudf_version=22.06.0-20220420.050427-10
user=
revision=aa3dbab35832bb8a70b3daa13cbbdc1171825180
branch=branch-22.06
date=2022-05-10T07:20:48Z
url=https://github.com/NVIDIA/spark-rapids.git
root@d03af8a1b975:/spark-rapids/dist/target/tmp#
root@d03af8a1b975:/spark-rapids/dist/target/tmp# cat spark-rapids-jni-version-info.properties
version=22.06.0-SNAPSHOT
user=
revision=854d2dd1619ddcc8514c80081c656dff166ff4d7
branch=HEAD
date=2022-04-20T04:58:01Z
url=https://github.com/NVIDIA/spark-rapids-jni.git
root@d03af8a1b975:/spark-rapids/dist/target/tmp#

```
Though they refer to the same spark-rapids-jni jar dependency, running integration tests still failed with cudf expected/actual version MISMATCH

```

22/05/10 06:25:28.740 ScalaTest-main-running-MortgageSparkSuite INFO RapidsExecutorPlugin: RAPIDS Accelerator build: {version=22.06.0-SNAPSHOT, user=, url=https://github.com/NVIDIA/spark-rapids.git, date=2022-05-10T06:25:13Z, revision=aa3dbab35832bb8a70b3daa13cbbdc1171825180, cudf_version=22.06.0-20220420.050427-10, branch=branch-22.06}

22/05/10 06:25:28.741 ScalaTest-main-running-MortgageSparkSuite INFO RapidsExecutorPlugin: cudf build: {version=22.06.0-SNAPSHOT, user=, url=https://github.com/rapidsai/cudf.git, date=2022-04-20T04:58:01Z, revision=65b1cbdeda9cab57243d0a98e646c860ef86039e, branch=HEAD}
22/05/10 06:25:28.743 ScalaTest-main-running-MortgageSparkSuite ERROR RapidsExecutorPlugin: Exception in the exec 
```

Reply:
> Could you please help to check if this change what we expect fix for the issue https://github.com/NVIDIA/spark-rapids/issues/5294?

It does not per the prior discussion.  This is a good step towards the fix, but this does not address the deduplication and manual unshimming required for non-Spark dependencies.

> Maybe need we also to add some test cases in the file?

Yes, test cases should be added for the new behavior for SNAPSHOT matching.